refgenconf/refgenconf.py

```diff
@@ -2087,11 +2164,148 @@ def _select_genomes(self, genome=None, strict=False, order=None,
 
 
+def config_upgrade(target_version, filepath, force=False):
+    """
+    upgrade the config file to a target version,
+    and alter genome_folder stucture
+
+    :param str target_version: the version updated to
+    :param bool force
+    """
+
+    # reformat config file
+
+    def format_config(target_version, get_json_url=lambda server: construct_request_url(server, API_ID_ALIAS_DIGEST)):
+        missing_digest = []
+        # check if any genome lack of local fasta asset and not on server
+        for genome, genome_v in rgc[CFG_GENOMES_KEY].items():
+            digest = ""
+            # get genome digest from the server
+            cnt = 0
+            servers = rgc[CFG_SERVERS_KEY]
+            for server in servers:
+                cnt += 1
+                url_alias = get_json_url(
+                    server=server).format(alias=genome)
+                try:
+                    digest = _download_json(url_alias)
+                except DownloadJsonError:
+                    if cnt == len(servers):
+                        try:
+                            tag = rgc.get_default_tag(genome, "fasta")
+                            asset_path = rgc.seek(
+                                genome, "fasta", tag, enclosing_dir=True)
+                            ssc = SeqColClient({})
+                            digest, _ = ssc.load_fasta(
+                                os.path.join(asset_path, genome+'.fa'))
+                        except:
+                            continue
+                    continue
+
+            if digest:
+                # create "aliases" section
+                genome_v[CFG_ALIASES_KEY] = [genome]
+                # convert seek keys, childran/parent asset keys from aliases to genome digests
+                for asset, asset_v in genome_v[CFG_ASSETS_KEY].items():
+                    for tag, tag_v in asset_v[CFG_ASSET_TAGS_KEY].items():
+                        for seek, seek_v in tag_v[CFG_SEEK_KEYS_KEY].items():
+                            asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_SEEK_KEYS_KEY][seek] = seek_v.replace(
+                                genome, digest)
+
+                        if (asset == "fasta" and CFG_ASSET_CHILDREN_KEY in asset_v[CFG_ASSET_TAGS_KEY][tag]):
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i].replace(
+                                    genome, digest)
+                        elif asset != "fasta":
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i].replace(
+                                    genome, digest)
```

Comment:
I don't get this part.. Why did you make the fasta asset a special case with these conditionals?